pkg/flow/deduper.go

```diff
@@ -79,17 +73,36 @@ func (c *deduperCache) isDupe(key *ebpf.BpfFlowId) bool {
 		c.entries.MoveToFront(ele)
 		// The input flow is duplicate if its interface is different to the interface
 		// of the non-duplicate flow that was first registered in the cache
-		return fEntry.ifIndex != key.IfIndex
+		// except if the new flow has DNS enrichment in this case we will
+		// mark the original flow as duplicate
+		if r.DNSLatency != 0 && fEntry.record.DNSLatency == 0 && fEntry.ifIndex != r.Id.IfIndex {
+			if justMark {
+				fEntry.record.Duplicate = true
+			} else {
+				fwd = findAndDeleteRecord(fwd, fEntry.record)
```

Comment:
This won't work if the entry was in cache but already sent previously (via a previous set of records): it won't be in the `fwd` slice.

Reply:
based on the info in the arch doc my understanding is the dup stage after the accout , plus of the record not in the same fwd slice in that case it won't be considered dup right ?

Reply:
I'm wrong about the accounter but IMO the problem is still the same: a flow could have been sent out to FLP and its dup comes in later.

> the record not in the same fwd slice in that case it won't be considered dup right ?

I think that's wrong and that's why there is a separate cache with its own expiry time: we keep track of flow keys in that cache, even for flows already sent to FLP, so that flows coming later can still be seen as duplicates